src/Julian.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roukmoute\Polyfill\Calendar;
+
+final class Julian
+{
+    const HALAKIM_PER_HOUR = 1080;
+    const HALAKIM_PER_DAY = 25920;
+    const HALAKIM_PER_LUNAR_CYCLE = (29 * self::HALAKIM_PER_DAY) + 13753;
+    const HALAKIM_PER_METONIC_CYCLE = self::HALAKIM_PER_LUNAR_CYCLE * (12 * 19 + 7);
+
+    const JEWISH_SDN_OFFSET = 347997;
+    const NEW_MOON_OF_CREATION = 31524;
+
+    const SUNDAY = 0;
+    const MONDAY = 1;
+    const TUESDAY = 2;
+    const WEDNESDAY = 3;
+    const THURSDAY = 4;
+    const FRIDAY = 5;
+    const SATURDAY = 6;
+
+    const NOON = 18 * self::HALAKIM_PER_HOUR;
+    const AM3_11_20 = (9 * self::HALAKIM_PER_HOUR) + 204;
+    const AM9_32_43 = (15 * self::HALAKIM_PER_HOUR) + 589;
+
+    const JULIAN_SDN_OFFSET = 32083;
+    const DAYS_PER_5_MONTHS = 153;
+    const DAYS_PER_4_YEARS = 1461;
+
+    private static $yearOffset = [
+        0, 12, 24, 37, 49, 61, 74, 86, 99, 111, 123,
+        136, 148, 160, 173, 185, 197, 210, 222,
+    ];
+
+    private static $monthsPerYear = [
+        12, 12, 13, 12, 12, 13, 12, 13, 12, 12,
+        13, 12, 12, 13, 12, 12, 13, 12, 13,
+    ];
+
+    /**
+     * @author treebe
+     * @see https://www.php.net/manual/en/function.jdtogregorian.php#32561
+     */
+    public static function jdtogregorian(int $julian_day): string
+    {
+        $julian = $julian_day - 1721119;
+        $calc1 = 4 * $julian - 1;
+        $year = floor($calc1 / 146097);
+        $julian = floor($calc1 - 146097 * $year);
+        $day = floor($julian / 4);
+        $calc2 = 4 * $day + 3;
+        $julian = floor($calc2 / 1461);
+        $day = $calc2 - 1461 * $julian;
+        $day = floor(($day + 4) / 4);
+        $calc3 = 5 * $day - 3;
+        $month = floor($calc3 / 153);
+        $day = $calc3 - 153 * $month;
+        $day = floor(($day + 5) / 5);
+        $year = 100 * $year + $julian;
+
+        if ($month < 10) {
+            $month = $month + 3;
+        } else {
+            $month = $month - 9;
+            $year = $year + 1;
+        }
+
+        return "$month/$day/$year";
+    }
+
+    /**
+     * @author Scott E. Lee
+     * @see https://github.com/php/php-src/blob/5b01c4863fe9e4bc2702b2bbf66d292d23001a18/ext/calendar/jewish.c
+     */
+    public static function jewishtojd(int $month, int $day, int $year): int
```

Comment:
I think it should go to `Jewish.php`, no?

Reply:
🤔 Indeed, either we group by output then `jdtogregorian` would go in Gregorian.php or by input and this one goes in Jewish.php, this also mean constants should be gathered in some other interface for sharing.